R/step_population_scaling.R

```diff
@@ -0,0 +1,193 @@
+#' Create a recipe step that scales variables using population data
+#'
+#' `step_population_scaling` creates a specification of a recipe step
+#' that will add a population scaled column in the data. For example,
+#' load a dataset that contains county population, and join to an `epi_df`
+#' that currently only contains number of new cases by county. Once scaled,
+#' predictions can be made on case rate.
+#'
+#' @param recipe A recipe object. The step will be added to the sequence of
+#' operations for this recipe. The recipe should contain information about the
+#' `epi_df` such as column names.
+#' @param df a data frame that contains the population data used for scaling.
+##' @param by A character vector of variables to join by. Column names should be
+#' in lower case.
+#' To join by different variables on the `epi_df` and `df`, use a named vector.
+#' For example, by = c("geo_value" = "states") will match `epi_df$geo_value`
+#' to `df$states`. To join by multiple variables, use a vector with length > 1.
+#' For example, by = c("geo_value" = "states", "county" = "county") will match
+#' `epi_df$geo_value` to `df$states` and `epi_df$county` to `df$county`.
```

Comment:
Same as above. Maybe also point to the `dplyr::full_join()` documentation.

Reply:
Good idea, I'll point to the the `dplyr::*_join()` document. I deliberately chose to use a `dplyr::left_join()` to avoid hassles of bringing in columns in the population data. Although we can always use `recipes::step_rm()` to remove the unwanted columns in the population data so it doesn't get mixed up in the modelling steps. I think I'll also make it clear in the documentation that a left join was performed.

Reply:
Great, thanks!